src/mbgl/style/style.cpp

```diff
@@ -328,6 +328,43 @@ RenderData Style::getRenderData() const {
     return result;
 }
 
+std::vector<std::string> Style::queryRenderedFeatures(
+        const std::vector<TileCoordinate>& queryGeometry,
+        const double zoom,
+        const double bearing,
+        const optional<std::vector<std::string>>& layerIDs) {
+    std::vector<std::unordered_map<std::string, std::vector<std::string>>> sourceResults;
```

Comment:
Nope. Features need to be returned in order based on the style order. Since layers from a source aren't all consecutive two steps are needed to get this sorting right.

added a short comment: 0d180639f92e3fc739472f33315b851b405ecd94